torch/csrc/autograd/engine.cpp

```diff
@@ -572,9 +581,11 @@ auto Engine::execute(const edge_list& roots,
   });
 
   // Callbacks are only valid for the duration of this run and should always be cleared
+  // Lock post_callbacks_lock_ before clearing final_callbacks_
   ClearCallbacks _cb_guard(final_callbacks_, post_callbacks_lock_);
 
   GraphTask graph_task(keep_graph, create_graph);
+  // For condition variable graph_task.not_done_
```

Comment:
We just created a new GraphTask, and then took out the lock on it. I don't think this has to do with CV; the function is to prevent threads for immediately banging on `graph_task` before we're done setting it up (e.g., with `init_to_execute`)